### DISCUSSION TITLE
Getting Socket Timeout Errors

### DISCUSSION THREAD

**Discussion by ghost** [nextgenhealthcare/connect#5677] [Q&A]
Hi All!

We've been facing a few problems with our Mirth 4.0.0 we've configured some channels but there are intermittent issues of `socket timeout`.

Additionally we configured 2 sftp channels which connect to the same client but on 2 different URLs however one of the channels works while the other just throws back a `socket timeout` error.

The possible issue is that we have 2 interfaces, one connects to the internet while the other one connects to VPNs.

One possible solution is to add a `BindAddress` directive to the `jsch custom configurations` in advanced options for SFTP.
But there's no documentation on that. 
Also I'd looked at one solution with `jmx.host` which would bind to a specific IP address, but can't find the directive in the docs for version 4.x.x.

Could someone help with this?
![Screenshot 2023-02-14 at 14 23 03](https://user-images.githubusercontent.com/101239358/218691397-7a911a6c-f174-4a90-b161-e75149917ae8.jpg)

Thanks and Regards,
Firdaus Aga

**Comment by ab-20-m** [comment]
In looking at the code for JSch, it does not appear the library supports easily binding to a specified local address. It looks like the only way to do so is specify a custom `SocketFactory` for the `Session` class to use and have the custom factory create the `Socket` bound to the local address desired.

Short term your best bet might be to increase the timeout value. I believe someone once did a Javascript Reader channel that used [Apache MINA SSHD](https://mina.apache.org/sshd-project/) to connect to a server and do something with SSH or SFTP. Probably a good deal of work and I think there were some classpath issues with version of Netty included with Mirth versus was needed for MINA SSHD.

**Reply by ghost** [reply]
Thanks for the quick response @ab-20-m!
Is there any way to set the bind address for the Mirth Service itself?

**Reply by ab-20-m** [reply]
The `http.host` and `https.host` properties in the `mirth.properties` file control what addresses the Mirth Service binds to for the API endpoints.

**Reply by ghost** [reply]
I've tried that and it works, but only for the `Administrator Launcher` not for the channel itself by the looks of it.

I'd read about some `jmx.host` or something but couldn't find the supporting config in the docs, is that still valid?
Or is there something similar that I could try?

**Reply by ab-20-m** [reply]
As far as I am aware, there's no way to specify the local bind address on sockets through a JVM property.

**Reply by ghost** [reply]
Got it, Should I create a feature request for the same?
